src/dns-resolve.ts

```diff
@@ -8,6 +8,8 @@ const lruOptions = { max: 500 };
 let cache4: LRU<string, string | Promise<string>>;
 let cache6: LRU<string, string | Promise<string>>;
 
+const localhostRegex = /localhost[.]?$/
```

Comment:
Should it begin with `^`?

Reply:
if I'm reading the RFC correctly https://tools.ietf.org/html/rfc6761#section-6.3

> The domain "localhost." and any names falling within ".localhost." are special in the following ways:

> 3. Name resolution APIs and libraries SHOULD recognize localhost names as special and SHOULD always return the IP loopback address for address